test/site/_transforms/service-worker-partials/index.js

```diff
@@ -0,0 +1,97 @@
+const assert = require("assert");
+const path = require("path");
+const cheerio = require("cheerio");
+const fs = require("fs").promises;
+const {
+  serviceWorkerPartials,
+  getPartial,
+  writePartial,
+} = require("../../../../src/site/_transforms/service-worker-partials");
+
+describe("service-worker-partials", function() {
+  let $;
+
+  beforeEach(function() {
+    const fixture = `<html lang="en"><head><title>Test title</title></head><body><div id="content"><div class="guide-landing-page"><div>Hello</div></div><div>Bonus sibling</div></div></body></html>`;
+    $ = cheerio.load(fixture);
+  });
+
+  afterEach(function() {
+    $ = null;
+  });
+
+  describe("serviceWorkerPartials", function() {
+    it("converts a string of html to a partial and writes it to disk", async function() {
+      const expected = {
+        raw: $("#content").html(),
+        lang: $("html").attr("lang"),
+        title: $("title").text(),
+        offline: false,
+      };
+
+      const hash = Math.random().toString(36);
+      const outputPath = path.join(".", ".tmp", hash, "index.html");
+      await serviceWorkerPartials($.html(), outputPath);
+      const actual = JSON.parse(
+        await fs.readFile(outputPath.replace(".html", ".json"), "utf-8"),
+      );
+      assert.deepStrictEqual(actual, expected);
+      // Cleanup .tmp dir
+      fs.rmdir(path.join(".", ".tmp"), {recursive: true});
```

Comment:
`await`

Actually, can we add this to the `afterEach` section? Even `fs.rmdirSync` if we can't await there.

```js
const fsBase = require("fs");
const {promises: fs} = fsBase;

describe('service-worker-partials', function() {

  afterEach(function() {
    $ = null;
    fsBase.rmdirSync(path.join(".", ".tmp"), {recursive: true});
  });
```